src/ts/components/core/checkbox/CheckboxCard.tsx

```diff
@@ -0,0 +1,87 @@
+    CheckboxIndicator,
+    Group,
+    MantineColor,
+    MantineRadius,
+    MantineSize,
+} from "@mantine/core";
+import { BoxProps } from "props/box";
+import { DashBaseProps, PersistenceProps } from "props/dash";
+import { StylesApiProps } from "props/styles";
+import React from "react";
+import { setPersistence, getLoadingState, applyDashProps } from "../../../utils/dash3";
+
+interface Props
+    extends BoxProps,
+        StylesApiProps,
+        DashBaseProps,
+        PersistenceProps {
+    /** Key of `theme.colors` or any valid CSS color to set input background color in checked state, `theme.primaryColor` by default */
+    color?: MantineColor;
+    /** Controls size of the component, `'sm'` by default */
+    size?: MantineSize | (string & {});
+    /** Determines whether the card should have border, `true` by default */
+    withBorder?: boolean;
+    /** Key of `theme.radius` or any valid CSS value to set `border-radius,` `theme.defaultRadius` by default */
+    radius?: MantineRadius;
+    /** Props passed down to the root element */
+    wrapperProps?: Record<string, any>;
+    /** Indeterminate state of the checkbox. If set, `checked` prop is ignored. */
+    indeterminate?: boolean;
+    /** Key of `theme.colors` or any valid CSS color to set icon color, by default value depends on `theme.autoContrast` */
+    iconColor?: MantineColor;
+    /** Determines whether icon color with filled variant should depend on `background-color`. If luminosity of the `color` prop is less than `theme.luminosityThreshold`, then `theme.white` will be used for text color, otherwise `theme.black`. Overrides `theme.autoContrast`. */
+    autoContrast?: boolean;
+    /** To be used with checkbox group */
+    value?: string;
+    /** State of check box */
+    checked?: boolean;
+    /** Whether component is disabled */
+    disabled?: boolean;
+    /** Icon */
+    icon?: React.ReactNode;
+    /** Indeterminate icon */
+    indeterminateIcon?: React.ReactNode;
+    /** Card content */
+    children?: React.ReactNode;
+}
+
+/** CheckboxCard */
+const CheckboxCard = ({
+                          children,
+                          setProps,
+                          loading_state,
+                          persistence,
+                          persisted_props,
+                          persistence_type,
+                          icon,
+                          indeterminateIcon,
+                          ...others
+                      }: Props) => {
+
+
+    const iconFunc = ({ indeterminate, ...others }) => {
+        const selected: any = indeterminate ? indeterminateIcon : icon;
+        return applyDashProps(selected, others);
+    };
+
+    return (
+        <MantineCheckboxCard
+            data-dash-is-loading={getLoadingState(loading_state) || undefined}
+            onChange={(state: boolean) => setProps({ checked: state })}
+            {...others}
+        >
+            <Group wrap="nowrap" align="flex-start">
```

Comment:
I like the convenience of having the `CheckboxCard` include the `CheckboxIndicator`, but it limits the ability to style the card.  For example it's not possible to to use something other than `Group` and no way to change these Group props.   

I think it will be better to have a stand-alone CheckboxCard, and people can build their own content and place the `CheckboxIndicator` and other content in the children prop.

Reply:
In hindsight it makes sense to simpify it, i was trying to implement the card from [here](https://mantine.dev/core/checkbox/#checkboxcard-component)